spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/RandomGenerator.scala

```diff
@@ -46,6 +46,10 @@ object RandomGenerator {
   }
 }
 
+/**
+ * A non-global random generator that carries its own state
```

Comment:
A mersenne twister based fake random number generator. Please refer https://en.wikipedia.org/wiki/Mersenne_Twister. Note that it has its own state so it is not thread safe. So you should use RandomGenerator.RNG to get a thread local instance to use. That's thread-safe.